packages/cadl-ranch-specs/http/lro/lro-core/main.cadl

```diff
@@ -0,0 +1,131 @@
+import "@azure-tools/cadl-azure-core";
+import "@azure-tools/cadl-ranch-expect";
+import "@cadl-lang/rest";
+
+using Azure.Core;
+using global.Azure.Core.Traits;
+using Cadl.Rest;
+using Cadl.Http;
+
+#suppress "@azure-tools/cadl-azure-core/casing-style" "For spec"
+@doc("Illustrates bodies templated with Azure Core with long-running operation")
+@scenarioService("/azure/lro/core")
+namespace Azure.Lro.Core;
+
+interface ResourceOperations
+  extends global.Azure.Core.ResourceOperations<NoConditionalRequests & NoRepeatableRequests & NoClientRequestId> {}
+
+@resource("users")
+@doc("Details about a user.")
+model User {
+  @key
+  @visibility("read")
+  @doc("The name of user.")
+  name: string;
+
+  @doc("The role of user")
+  role: string;
+}
+
+@doc("The parameters for exporting a user.")
+model UserExportParams {
+  @query
+  @doc("The format of the data.")
+  format: string;
+}
+
+@doc("The exported user data.")
+model ExportedUser {
+  @doc("The name of user.")
+  name: string;
+
+  @doc("The exported URI.")
+  resourceUri: string;
+}
+
+@scenario
+@doc("Creates or repalces a User")
+@summary("Adds a user or repalces a user's fields.")
+@scenarioDoc("""
+Should only generate one model named User.
+
+Expected path parameter: name=madge
+
+Expected input body:
+```json
+{
+  "role": "contributor"
+}
+```
+
+Expected response body:
```

Comment:
The response of an LRO PUT should the resource -- a User in this case.

Reply:
Yes, the final result is the User resource.

The first 2 in expected responses describes the polling operation. For Java, these poll result also visible to user, in [SyncPoller/PollerFlux](https://azure.github.io/azure-sdk/java_introduction.html#java-lro-poller-class) (the `T` type)

Reply:
I've enhance the `@scenarioDoc` to spell out details on every request/response.